docs/src/pages/components/passwordfield/react.mdx

```diff
@@ -207,69 +171,42 @@ Use the `isRequired` prop to specify a required field.
 ```
 
 <Example>
-  <Flex as="form" direction="column" width="20rem">
-    <PasswordField label="Password" name="password" isRequired={true} />
-    <Button type="submit">Submit</Button>
-  </Flex>
+  <IsRequiredExample />
+  <ExampleCode>
+    
+    ```jsx file=./examples/IsRequiredExample.tsx
+
+    ```
+
+  </ExampleCode>
 </Example>
 
 There is no default styling for required fields. Customize the `label` or `descriptiveText` to instruct the form user of the required field.
 
-```jsx
-<Flex as="form" direction="column" width="20rem">
-  <PasswordField
-    label={
-      <Text>
-        Password
-        <Text as="span" fontSize="0.8rem" color="red">
-          {' '}
-          (required)
-        </Text>
-      </Text>
-    }
-    name="password"
-    isRequired={true}
-  />
-  <PasswordField
-    label="Password"
-    descriptiveText={
-      <Text as="span" fontSize="0.8rem" color="red" fontStyle="italic">
-        Required
-      </Text>
-    }
-    name="password"
-    isRequired={true}
-  />
-  <Button type="submit">Submit</Button>
-</Flex>
-```
-
 <Example>
   <RequiredPasswordFieldExample />
+  <ExampleCode>
+
+    ```jsx file=./examples/RequiredPasswordFieldExample.tsx
+
+    ```
+
+  </ExampleCode>
 </Example>
 
 ### Validation error styling
 
 Use the `hasError` and `errorMessage` fields to mark a PasswordField as having an validation error.
 
-```jsx
-<PasswordField
-  label="Password"
-  defaultValue="1234"
-  hasError={true}
-  errorMessage="Password should not be 1234! 😱"
-/>
-```
-
 <Example>
-  <Flex gap="1rem" direction="column">
-    <PasswordField
-      label="Password"
-      defaultValue="1234"
-      hasError={true}
-      errorMessage="Password should not be 1234! 😱"
-    />
-  </Flex>
+  <ValidationErrorExample />
+  <ExampleCode>
+
+    ```jsx file=./examples/ValidationErrorExample.tsx
+
+    ```
+
+  </ExampleCode>
 </Example>
 
 ### Forward refs
```

Comment:
Perhaps we should briefly explain what the customer should expect from the code below. For example, something like: "In the example below, when you click on the showPassword button, the input element becomes focused."